### PR TITLE
feat/fix(fees): fix fee claiming by updating position accumulator value to fee growth outside

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Discovered by @czarcas7ic 

Previously, when we claimed the fees, we used the default claim logic that does not account for custom range positions.

With concentrated liquidity, accumulator positions track fee growth outside. As a result, a new method `ClaimRewardsCustomAcc` is introduced in the `accum` package in this PR.

Additionally, tests added for `CollectFees` to ensure that the position fee accumulator is not updated to the global fee growth value. 

To implement this test, a new method `GetPositionValue` was introduced in the `accum` package.

## Brief Changelog

- implement and test `ClaimRewardsCustomAcc` 
- fix the bug in `CollectFees` by calling the new claim function
- implement and test `GetPositionValue`
- use `GetPositionValue` to test that the position accumulator in `CollectFees` is not equal to the global fee growth accumulator.

## Testing and Verifying

This change added tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/concentrated-liquidity/fees.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ func (k Keeper) collectFees(ctx sdk.Context, poolId uint64, owner sdk.AccAddress
 	}
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/fees.go:214]: I believe the lines above this (210-213) need to be removed. We are properly accounting for the current fee growth outside when we use the ClaimRewardsCustomAcc func, so updating the accumulator prior that causes issues.

**Comment by czarcas7ic** [comment]
Will describe in today's call why this needs further changes to completely work. Deleted my previous in line comments since the problem was involving more than what I just pointed out.

**Comment by czarcas7ic** [comment]
Here is the commit I made on my branch with all the fixes https://github.com/osmosis-labs/osmosis/commit/b45eb46a5477dc55136868a12e4a1de404dd6748

**Comment by p0mvn** [comment]
synched offline - pursuing another direction for fixes

**Pull request closed without merging**